subsys/net/lib/openthread/platform/radio.c

```diff
@@ -302,6 +306,41 @@ static void openthread_handle_received_frame(otInstance *instance,
 	net_pkt_unref(pkt);
 }
 
+static void openthread_handle_frame_to_send(otInstance *instance,
+					    struct net_pkt *pkt)
+{
+	struct net_buf *buf;
+	otMessage *message;
+	otMessageSettings settings;
+
+	NET_DBG("Sending Ip6 packet to ot stack");
+
+	settings.mPriority = OT_MESSAGE_PRIORITY_NORMAL;
+	settings.mLinkSecurityEnabled = true;
+	message = otIp6NewMessage(instance, &settings);
+	if (message == NULL) {
+		goto exit;
+	}
+
+	for (buf = pkt->buffer; buf; buf = buf->frags) {
+		if (otMessageAppend(message, buf->data,
+				    buf->len) != OT_ERROR_NONE) {
+
+			NET_ERR("Error while appending to otMessage");
+			otMessageFree(message);
+			goto exit;
+		}
+	}
+
+	if (otIp6Send(instance, message) != OT_ERROR_NONE) {
+		NET_ERR("Error while calling otIp6Send");
```

Comment:
I see it was like this before, but I wonder - will `otIp6Send` free the message in case of error? We don't call `otMessageFree(message)` here, could you verify that?

Reply:
Yes, otIp6Send calls free if it fails to send the message.